lib/pages/activities/activity_page.dart

```diff
@@ -72,12 +72,35 @@ class ActivityPage extends StatelessWidget {
                 itemCount: _activities.length,
                 itemBuilder: (context, index) {
                   final activity = _activities[index];
-                  return ActivityCard(
-                    taskName: activity.taskName,
-                    courseName: activity.courseName,
-                    remainingTime: activity.remainingTime,
-                    topicTheme: activity.topicTheme,
+                  return InfoCard(
                     backgroundColor: activity.backgroundColor,
+                    children: [
+                      InfoCardItem(
+                        child: Text(
+                          activity.taskName,
+                          style:
+                              textTheme.headlineSmall?.withColor(Colors.white),
+                        ),
+                      ),
```

Comment:
This could be replaced with the `title` property of `InfoCard` that already applies the `headlineSmall` text style with the correct `foregroundColor` (either the calculated or the explicit one).

```suggestion
                  return InfoCard(
                    backgroundColor: activity.backgroundColor,
                    title: Text(activity.taskName),
                    children: [
```